### ISSUE TITLE
[Bug] Paragraph with tag not queryable, if it is on the first line

### ISSUE THREAD

**Issue by buxel** [silverbulletmd/silverbullet#586]
Hi! 
I was trying to aggregate some tagged notes from daily notes using tags when i noticed some of them were not returned by my query.

```query
paragraph 
where tags = "TagA"
```

After some investigation it seems like if a tagged paragraph is on the first line of a document, it is not returned:

![image](https://github.com/silverbulletmd/silverbullet/assets/331779/72a9a15e-5427-4768-abc8-0f9a4ddc8c49)

**Comment by zefhemel** [comment]
Yeah, this is by design and I'm thinking what to do here. The reason is that the first paragraph is "reserved" for tags that should be attached to the page as a whole. If I'd also index it as a paragraph, querying `TagA` (not sure what your query here looks like) would return both the `page` and `paragraph` items, which would be quite messy.

Suggestions?

**Comment by buxel** [comment]
Would it be possible to have page-wide tags (and maybe other things) in frontmatter? Then it would be clearly separated from the content and not hidden functionality in the first line.

**Comment by zefhemel** [comment]
Yes, but that would mean the only way to tag pages would be to use frontmatter. That's an option, but perhaps not super convenient for people. If you have a page that you wouldn't normally put frontmatter in (or if you don't even know what frontmatter is), you could just throw in a hashtag to tag it — you'd lose that ability then.

**Comment by buxel** [comment]
Fair. I like how simplistic the current solution is but it does not feel intuitive and "clean". 
It seems like a shortcut to what frontmatter is supposed to do. A shortcut, that breaks the paragraph tagging functionality in a non-obvious way.

My thinking is: 
1. The scope "page" is one level higher than its "content". 
1. Within the "content" scope, it is intuitive to tag and query 
1. The "page" scope is somehow abstract and wedged into the "content" scope. A compromise has to be made somehow.

IMO, the technically correct way (and we all know this is the best way of correct 😉 ) is to store tags in frontmatter because it keeps the scopes separated.
So what about someone who just wants to tag a page easily? How can it still be simple to add tags (or other metadata) to pages? While I'm not an expert in that area, I think this is more of an UX problem and could be solved with a dedicated frontmatter editor at he top / to the side. For starters, it could allow you to only edit a subset of frontmatter (the tags).

**Comment by buxel** [comment]
After writing all that, another option crossed my mind: prefix all page-wide tags i.e. with `#page::tempalte` ?
I don't like it so much but it wins in simplicity

**Comment by zefhemel** [comment]
Yeah one option is to indeed just support page-level tagging in frontmatter, that would be the simple solution.

The trouble started when somebody implemented `paragraph` indexing (and with that: paragraph tagging). I'm still not super clear on the use cases of `paragraph` myself (I don't use it myself). Before that this wasn't ambiguous, the initial implementation even allowed tags in any top-level paragraph in the page, and would attach those tags to the page. That seemed a bit excessive too.

🤔

**Comment by zefhemel** [comment]
From the UX perspective, it'd be trivial to add a "Tag: Page" or similar command and just add the frontmatter automatically. Commands do have a big discoverability issue, though.

**Comment by zefhemel** [comment]
Thinking of the "how many people would be affected" angle: I'd expect the number of people using paragraph level queries to be extremely low, whereas the number of people who'd like to throw in a quick `#tag` to tag a page would be quite high. Fair?

**Comment by buxel** [comment]
Maybe I'm thinking about this wrong... My usecase is:
- have daily notes, mostly random thoughts.  Maybe part of a project
- i jot everythign down and tag it with #idea, #project-a or #follow-up
- i would like some queries to redeiscover those notes/thoughts

Is there any other way than doing it with paragraph tagging? To me it is important to not have to think about where to put some information and i can just quickly dump it into daily notes and refine it later. For that, it needs to be somehow discoverable, as i will not go trough every daily note one by one.

**Comment by zefhemel** [comment]
This should now be fixed with https://github.com/silverbulletmd/silverbullet/commit/848211120c2d8fc0b9c81c0c1e5c8bff3b3bcb8b (the semantics around page tagging have changed, see the CHANGELOG)
